common-util/Contracts/index.jsx

```diff
@@ -93,7 +102,13 @@ export const getTokenomicsContract = () => {
 };
 
 export const getUniswapV2PairContract = (address) => {
-  const contract = getContract(UNISWAP_V2_PAIR_ABI, address);
+  const { chainId } = getWeb3Details();
+
+  // TODO: check if this is correct @kupermind
+  const contract = getContract(
+    chainId === 1 ? UNISWAP_V2_PAIR_ABI : GNOSIS_BRIDGE_PROXY,
```

Comment:
As of now, we only need to use the UNISWAP pair ABI, as that's what we are going to get on the gnosis chain as well